bsmetadata/metadata_utils.py

```diff
@@ -124,16 +124,27 @@ def is_metadata(idx: int) -> bool:
         for text_chunk_encoded, chunk_metadata_mask in chunks(
             max_text_len, text_with_local_metadata_encoded.input_ids, token_level_metadata_mask
         ):
-            total_len = prefix_len + len(text_chunk_encoded)
-            padding_len = max_text_len - len(text_chunk_encoded)
-
-            input_ids = metadata_prefix_encoded + text_chunk_encoded + [tokenizer.eos_token_id] * padding_len
-            attention_mask = [1] * total_len + [0] * padding_len
-            metadata_mask = [1] * prefix_len + [int(x) for x in chunk_metadata_mask] + [0] * padding_len
-
-            linearized_examples["input_ids"].append(input_ids)
-            linearized_examples["attention_mask"].append(attention_mask)
-            linearized_examples["metadata_mask"].append(metadata_mask)
+            if without_metadata:
+                total_len = len(text_chunk_encoded)
+                padding_len = cfg.max_seq_len - len(text_chunk_encoded)
+                attention_mask = [1] * total_len + [0] * padding_len
+                text_chunk_encoded = text_chunk_encoded + [tokenizer.eos_token_id] * padding_len
+                input_ids = text_chunk_encoded + [tokenizer.eos_token_id] * padding_len
```

Comment:
`[tokenizer.eos_token_id] * padding_len` is added twice.
Is this a copy-paste mistake?

Reply:
you are right! Good spot.